artiq/coredevice/comm_moninj.py

```diff
@@ -21,8 +17,21 @@ class TTLOverride(Enum):
     oe = 2
 
 
-class CommMonInj:
+class ICommMonInj:
+    def monitor_probe(self, enable: int, channel: int, probe: int): raise NotImplementedError
+
+    def monitor_injection(self, enable: int, channel: int, overrd: int): raise NotImplementedError
+
+    def inject(self, channel: int, override: int, value: int): raise NotImplementedError
+
+    def get_injection_status(self, channel: int, override: int): raise NotImplementedError
+
+
+class CommMonInj(ICommMonInj):
     def __init__(self, monitor_cb, injection_status_cb, disconnect_cb=None):
+        self._reader = None
+        self._writer = None
+        self.endian = None
```

Comment:
I'm not sure if any of the modifications to this entire file is necessary or helpful.

Reply:
there is a problem regarding something like `Instance attribute <var name> defined outside __init__`, and I think it is worth it to clarify the variables

Reply:
What problem exactly?
Wasn't this error caused by ABC, which is now removed?

Reply:
well, intellij is complaining about it as a warning, and i think it is also weird that some variables exists only after you called some method, but anyway just keep it that way, i.e. not gonna set those variables in __init__

Reply:
IMO it's just a matter of taste and IntelliJ's opinion is as good as anyone else's. If you really want to set them to ``None`` in ``__init__``, I'm fine with that, but this is a style change and it does not belong in this PR. Please open another one.